File: src/main/java/com/zuhlke/rob/PianoSampleFactory.java
package com.zuhlke.rob;

import java.io.File;

public class PianoSampleFactory {
    private final File pianoSamplesDirectory;

    public PianoSampleFactory(File pianoSamplesDirectory) {
        this.pianoSamplesDirectory = pianoSamplesDirectory;
    }

    public PianoSample create(String noteSpecification) {
        return new PianoSample(new File(pianoSamplesDirectory, noteSpecification + ".aiff"));
    }
}

File: src/main/java/com/zuhlke/rob/PianoSampleFactory.kt
package com.zuhlke.rob

import java.io.File

class PianoSampleFactory(private val pianoSamplesDirectory: File) {
    fun create(noteSpecification: String): PianoSample {
        return PianoSample(File(pianoSamplesDirectory, "$noteSpecification.aiff"))
    }
}
